### PR TITLE
Update compileSdkVersion and build tools to 24

### PR DESCRIPTION
**Pull request by zaki50**
refs #3051

### CODE REVIEW

**Comment by zaki50** [comment]
@realm/java Should we keep `targetSdkVersion` of `realm-library` to `23` in `releases` branch for better compatibility to `1.0.1`?

File: examples/build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
-project.ext.sdkVersion = 23
-project.ext.buildTools = '23.0.0'
+project.ext.sdkVersion = 24
```
Comment (dalinaum) [inline examples/build.gradle]: Because most devices cannot support SDK version 24 yet, I don't know that using SDK version 23 is a good idea.

Comment (cmelchior) [inline examples/build.gradle]: The SDK version should not have any effect on what is actual running on the device. This is just used to compile everything with.

**Comment by cmelchior** [comment]
@zaki50 Do you know of any problems between 23.0.0 and 24.0.0 that might effect us? Otherwise I would say it should be safe to upgrade.

**Comment by kneth** [comment]
Yes, I think we should keep `23` in `releases`.

**Comment by zaki50** [comment]
@cmelchior I don't think that effects for us. But it has some effect for the users.
Realm's `targetSdkVersion` is merged into app's AndroidManifest.xml and then app's behavior might change.

I'll keep `targetSdkVersion` to `23` on `releases` branch.

**Comment by cmelchior** [comment]
Ah yes, good point 👍

File: Dockerfile [hunk 2]
```diff
@@ -30,10 +30,11 @@ RUN cd /opt && tar -xvzf android-sdk.tgz
 RUN cd /opt && rm -f android-sdk.tgz
 
 # Grab what's needed in the SDK
+RUN echo y | android update sdk --no-ui --all --filter tools > /dev/null
```
Comment (emanuelez) [inline Dockerfile]: can you explain why this different pattern to install `tools`? Maybe a comment will help not changing it to be like the others by mistake.

Comment (zaki50) [inline Dockerfile]: Right.

The story is a bit weird.

Running `RUN echo y | android update sdk --no-ui --all --filter tools` produces following output.
```
...
Installing Archives:
  Preparing to install archives
  Skipping 'Android SDK Tools, revision 25.1.7'; it depends on 'Android SDK Platform-tools, revision 24' which was not installed.
  Done. Nothing was installed.
```

This says that install was skipped and nothing was installed. But, if I remove this line, the lint check does not pass.

@emanuelez Do you know why this happens?

Comment (cmelchior) [inline Dockerfile]: They made some changes to the command line tool in the last versions, it might not even work anymore....

**Comment by kneth** [comment]
LGTM

**Comment by zaki50** [comment]
review this please @realm/java

**Comment by dalinaum** [comment]
LGTM

**Comment by cmelchior** [comment]
👍

**Pull request merged**